src/api.v2/events/handleWorkRequest.js

```diff
@@ -1,20 +1,41 @@
 const getLogger = require('../../utils/getLogger');
+const { OK } = require('../../utils/responses');
+const getWorkResults = require('../helpers/worker/getWorkResults');
 
 const logger = getLogger();
 const validateAndSubmitWork = require('./validateAndSubmitWork');
-const { authenticationMiddlewareSocketIO, authorize } = require('../middlewares/authMiddlewares');
 
 const handleWorkRequest = async (socket, data) => {
-  const { uuid, Authorization, experimentId } = data;
+  const { Authorization, experimentId, ETag } = data;
+
+  try {
+    const { signedUrl } = await getWorkResults(experimentId, ETag);
+
+    socket.emit(`WorkResponse-${ETag}`, {
+      request: { ...data },
+      results: [],
+      response: {
+        cacheable: false,
+        signedUrl,
+        error: null,
+      },
+    });
+
+    return;
+  } catch (e) {
+    if (e.status === 404) {
+      console.log(`Work result ${ETag} not cached, submitting request`);
+    } else {
+      throw e;
+    }
+  }
 
   try {
     // Authenticate and authorize the user
     if (!Authorization) {
       throw new Error('Authentication token must be present.');
     }
-    const jwtClaim = await authenticationMiddlewareSocketIO(Authorization, socket);
-    const { sub: userId } = jwtClaim;
-    await authorize(userId, 'socket', null, experimentId);
```

Comment:
why is this no longer necessary?

Reply:
Because the work requests don't arrive from a socket, but rather from an http endpoint which already has the authorization middleware running over it

Reply:
got it, thanks!